Gem/Code/Source/Components/NetworkCharacterComponent.cpp

```diff
@@ -182,6 +183,16 @@ namespace MultiplayerSample
 
     AZ::Vector3 NetworkCharacterComponentController::TryMoveWithVelocity(const AZ::Vector3& velocity, float deltaTime)
     {
+        // Ensure any entities that we might interact with are properly synchronized to their rewind state
```

Comment:
this chunk of code is showing up in a lot of places, makes me think we can make it a util function. :P

Reply:
Yup, it's pretty important. Currently it shows up in two places, but it will show up anywhere where we need to sync rewound entity state with an external library, like physx or recast for pathfinding, etc.. All it does is compute the AABB of the area you're about to interact with, and then call into the network time system to find any potential overlapping entities to tell them to sync.

Reply:
👍